src/main/java/org/wso2/carbon/module/swiftiso20022/mt/format/validators/MTBlockFormatValidator.java

```diff
@@ -0,0 +1,183 @@
+/**
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ * <p>
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.module.swiftiso20022.mt.format.validators;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.module.swiftiso20022.constants.ConnectorConstants;
+import org.wso2.carbon.module.swiftiso20022.constants.MTParserConstants;
+import org.wso2.carbon.module.swiftiso20022.mt.models.fields.Field111;
+import org.wso2.carbon.module.swiftiso20022.mt.models.fields.Field121;
+import org.wso2.carbon.module.swiftiso20022.validation.common.ValidationResult;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Class to validate MT message User Header Block and Trailer Block format.
+ */
+public class MTBlockFormatValidator {
+
+    private static final Log log = LogFactory.getLog(MTBlockFormatValidator.class);
+
+    /**
+     * This regex captures global match in the format -> one or more of {tag:value} .
+     * tag -> 3 digits
+     * value -> one or more character except {,},: and line break
+     */
+    private static final String USER_HEADER_BLOCK_FORMAT_REGEX = "^(\\{\\d{3}:[^}]{1,36}})+$";
```

Comment:
some of the regex patterns are defined in a separate class while some are as class  variables. What's the reason?

Reply:
These regexes are specifc regexes which can only be used for format validation. So there is no need for them to be present publicly.